lisa/analysis/tasks.py

```diff
@@ -256,11 +256,52 @@ def stringify_row_state(row):
 
         task_state_df.rename(columns={'__cpu' : 'cpu'}, inplace=True)
         task_state_df = task_state_df[['target_cpu', 'cpu', 'curr_state']]
-        task_state_df['next_state'] = task_state_df.curr_state.shift(-1)
+        # Make the last "next_state" the same as the last "curr_state"
+        task_state_df['next_state'] = task_state_df.curr_state.shift(
+            -1, fill_value=task_state_df['curr_state'].values[-1])
         self.trace.add_events_deltas(task_state_df, inplace=True)
 
         return task_state_df
 
+    def df_task_states_stringify(self, df_task_states, inplace=False):
```

Comment:
The name would benefit from:
* get an "internal" name prefixed by `_`, since it's only used by a similarly-named method.
* be turned into static/classmethod to make it explicit it does not touch the instance itself but is just a helper

Reply:
>  get an "internal" name prefixed by `_`, since it's only used by a similarly-named method.

I wanted to have it exposed publicly to make debugging things easier - if some tests is doing a lot of operations of a `task_states` dataframe, you could just do a `log.debug(df_task_states_stringify(df))` instead of trying to redo the same operations with the `_stringified` version.

Reply:
makes sense, in that case we could have it as a standalone function like `stringify_task_states_df` that is not especially related to a class, and more importantly that does not use `df_` prefix that is by convention reserved for "regular" `df_` functions that extract a df from the current trace (so that autocomplete suggestions are relevant in notebooks).